clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3472,14 +3460,15 @@ bool Util::isPropertyListType(const QualType &Ty) {
 }
 
 bool Util::isSyclBufferLocationType(const QualType &Ty) {
-  const StringRef &Name = "buffer_location";
+  const StringRef &PropertyName = "buffer_location";
+  const StringRef &InstanceName = "instance";
   std::array<DeclContextDesc, 6> Scopes = {
-      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "cl"},
-      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "sycl"},
-      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "ext"},
-      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "INTEL"},
-      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "property"},
-      Util::DeclContextDesc{Decl::Kind::ClassTemplateSpecialization, Name}};
+    Util::DeclContextDesc{Decl::Kind::Namespace, "cl"},
```

Comment:
How does this work with SYCL 2020 `::sycl::` instead of `::cl::sycl::`?

Reply:
@keryell it should be fine. We define `cl` as `inline namespace`.